test/normalize_tests.c

```diff
@@ -17,6 +19,22 @@ const char* tests[] = {
   "select count($1), date_trunc($2, created_at at time zone $3), $4, $5 from users group by date_trunc($2, created_at at time zone $3), date_trunc($6, created_at), $4, $5",
   "SELECT CAST('abc' as varchar(50))",
   "SELECT CAST($1 as varchar(50))",
+  "CREATE OR REPLACE FUNCTION pg_temp.testfunc(OUT response \"mytable\", OUT sequelize_caught_exception text) RETURNS RECORD AS $func_12345$ BEGIN INSERT INTO \"mytable\" (\"mycolumn\") VALUES ('myvalue') RETURNING * INTO response; EXCEPTION WHEN unique_violation THEN GET STACKED DIAGNOSTICS sequelize_caught_exception = PG_EXCEPTION_DETAIL; END $func_12345$ LANGUAGE plpgsql; SELECT (testfunc.response).\"mycolumn\", testfunc.sequelize_caught_exception FROM pg_temp.testfunc(); DROP FUNCTION IF EXISTS pg_temp.testfunc();",
+  "CREATE OR REPLACE FUNCTION pg_temp.testfunc(OUT response \"mytable\", OUT sequelize_caught_exception text) RETURNS RECORD AS $1 LANGUAGE plpgsql; SELECT (testfunc.response).\"mycolumn\", testfunc.sequelize_caught_exception FROM pg_temp.testfunc(); DROP FUNCTION IF EXISTS pg_temp.testfunc();",
+  "CREATE PROCEDURE insert_data(a integer, b integer) LANGUAGE SQL AS $$ INSERT INTO tbl VALUES (a); INSERT INTO tbl VALUES (b); $$",
+  "CREATE PROCEDURE insert_data(a integer, b integer) LANGUAGE SQL AS $1",
+  "DO $$DECLARE r record; BEGIN FOR r IN SELECT table_schema, table_name FROM information_schema.tables WHERE table_type = 'VIEW' AND table_schema = 'public' LOOP EXECUTE 'GRANT ALL ON ' || quote_ident(r.table_schema) || '.' || quote_ident(r.table_name) || ' TO webuser'; END LOOP; END$$",
+  "DO $1",
+  "CREATE SUBSCRIPTION mysub CONNECTION 'host=192.168.1.50 port=5432 user=foo dbname=foodb' PUBLICATION mypublication, insert_only",
+  "CREATE SUBSCRIPTION mysub CONNECTION $1 PUBLICATION mypublication, insert_only",
+  "ALTER SUBSCRIPTION mysub SET PUBLICATION insert_only",
+  "ALTER SUBSCRIPTION mysub SET PUBLICATION insert_only",
+  "ALTER SUBSCRIPTION mysub CONNECTION 'host=192.168.1.50 port=5432 user=foo dbname=foodb'",
+  "ALTER SUBSCRIPTION mysub CONNECTION $1",
+  "CREATE USER MAPPING FOR bob SERVER foo OPTIONS (user 'bob', password 'secret')",
+  "CREATE USER MAPPING FOR bob SERVER foo OPTIONS (user $1, password $2)",
+  "ALTER USER MAPPING FOR bob SERVER foo OPTIONS (SET password 'public')",
+  "ALTER USER MAPPING FOR bob SERVER foo OPTIONS (SET password $1)",
```

Comment:
Not blocking, but one might expect "mysub", "mypublication", "bob", and "foo" to be normalized in the above, no?

Reply:
Generally we've normalized constant values but not object references - and this follows that same intent.

Reply:
Got it, makes sense.